### ISSUE TITLE
Linux?

### ISSUE THREAD

**Issue by naxil** [sam-ludlow/mame-ao#22]
Any chance to have it under linux? Native?

**Comment by sam-ludlow** [comment]
Hello @naxil 

When you say native what we talking C/C++ applications or just recent .net?

I've have been thinking about this. Here are some thoughts.

I believe @cewatkins has had it running on WINE.

I made it framework 4.8 so it will run on most Windows boxes without requiring any other dependencies, like .net runtimes.

It should be pretty simple to port to current .net and run natively on anything that supports it like Linux and Pi.

The UI is just JS so should run on any browser. I didn't use any JS frameworks partly as an fun exercise, I don't really know or like any, and to keep it light weight.

But anyone could make an alternate UI the REST API is all in place ready to consume by anything.

Cheers
Sam

**Comment by sam-ludlow** [comment]
May just run on Mono with some minor code changes?

Be nice to only have a single code base.

**Comment by NikLP** [comment]
Current state of .Net on Linux (Ubuntu at least):

https://devblogs.microsoft.com/dotnet/whats-new-for-dotnet-in-ubuntu-2404/

TL;DR - Version 8

**Comment by cewatkins** [comment]
It works out of the box with wine.
I didn't send the update in as changes, as it's a little squirelly, but you can also run the server that way, with NTFS or NFS as the backend.
And, as far as symbolic links go, good luck.

**Comment by cewatkins** [comment]
Also, iirc, if you do things like add mono, especially it breaks it and it's not entirely obvious.

**Comment by cewatkins** [comment]
Oh and I forgot this one too.. 
Trying ::1...
Connection failed: Connection refused
Trying 127.0.0.1...
Connected to localhost.

NO V6.

**Comment by cewatkins** [comment]
This is proper:
[oo@oous mameao]$ wine mame-ao.exe 
01f0:err:winediag:gnutls_process_attach failed to load libgnutls, no support for encryption
01f0:fixme:ntdll:NtQuerySystemInformation info_class SYSTEM_PERFORMANCE_INFORMATION
1.87
'##::::'##::::'###::::'##::::'##:'########:::::::'###:::::'#######::
 ###::'###:::'## ##::: ###::'###: ##.....:::::::'## ##:::'##.... ##:
 ####'####::'##:. ##:: ####'####: ##:::::::::::'##:. ##:: ##:::: ##:
 ## ### ##:'##:::. ##: ## ### ##: ######::::::'##:::. ##: ##:::: ##:
 ##. #: ##: #########: ##. #: ##: ##...::::::: #########: ##:::: ##:
 ##:.:: ##: ##.... ##: ##:.:: ##: ##:::::::::: ##.... ##: ##:::: ##:
 ##:::: ##: ##:::: ##: ##:::: ##: ########:::: ##:::: ##:. #######::
..:::::..::..:::::..::..:::::..::........:::::..:::::..:::.......:::

       Please wait the first time it has to prepare the data

         The Web User Interface will pop up when ready

              See the README for more information
             https://github.com/sam-ludlow/mame-ao

######################################################################################################################################
#                                                            Initializing                                                            #
######################################################################################################################################
0208:fixme:file:server_get_file_info Unsupported info class e
0208:fixme:file:server_get_file_info Unsupported info class e
!!! You can save a lot of disk space by enabling symbolic links, see the README.
01f0:fixme:nls:GetFileMUIPath stub: 0x10, L"C:\\windows\\system32\\tzres.dll", (null), 0053ECB8, 0035DA70, 0053ECBC, 0053ECB0
01f0:fixme:nls:GetFileMUIPath stub: 0x10, L"C:\\windows\\system32\\tzres.dll", (null), 0053ECB8, 0035E248, 0053ECBC, 0053ECB0
01f0:fixme:nls:GetFileMUIPath stub: 0x10, L"C:\\windows\\system32\\tzres.dll", (null), 0053ECB8, 0035E248, 0053ECBC, 0053ECB0
MameVersion: 0265
======================================================================================================================================
=                                                                MAME                                                                =
=                           https://github.com/mamedev/mame/releases/download/mame0265/mame0265b_64bit.exe                           =
======================================================================================================================================
01f0:err:winediag:process_attach failed to load libgnutls, no support for pfx import/export
01f0:fixme:file:NtLockFile I/O completion on lock not implemented yet
Creating machine performance caches......done
Creating software performance caches......done
Getting all database SHA1s......done.
======================================================================================================================================
=                                                           Machine Genres                                                           =
=                      https://raw.githubusercontent.com/AntoPISA/MAME_SupportFiles/main/catver.ini/catver.ini                       =
======================================================================================================================================
Loading Genres......done
Version:        0.265
======================================================================================================================================
=                                                          Machine Samples                                                           =
=                        https://raw.githubusercontent.com/AntoPISA/MAME_Dats/main/MAME_dat/MAME_Samples.dat                         =
======================================================================================================================================
Version:        0.265

######################################################################################################################################
#                                                 Remote Listener ready for commands                                                 #
#                                                      http://localhost:12380/                                                       #
#                                   e.g. http://localhost:12380/api/command?line=a2600 et -window                                    #
######################################################################################################################################

01f0:fixme:exec:SHELL_execute flags ignored: 0x00004100
######################################################################################################################################
#                                                      Shell ready for commands                                                      #
######################################################################################################################################

**Comment by cewatkins** [comment]
Also, another comment, you can change it to run mame not mame.exe, but you have to run that
from 0265 and it works nicely, but you must have downloaded them first.
Also, this enables one, and also, it works with remote desktop as well, and it aint too sluggy.
I probably should have made more notes, but I'm not a check-iner really.
Also, I did try and ^M vs ^J and the two lines made 2600 changes, I was able to recompile
the changes on msvc (whatever it's called now, but the checkout and return process accross platforms is still funny.)
As far as I now, I probably could have put up a modded binary, but for remote connections with the server running it was the way mono/whatever m$ thing is called has a * version for localhost that is problem matic.

**Comment by sam-ludlow** [comment]
I'm not planning on porting at the moment.

Any small fixes to make it run better in Wine then I'd happily consider.

Porting from legacy .net framework to current .net on Linux should be pretty simple, but I'm not doing it anytime soon. This is probably the way to go but I don't want to maintain 2 codes bases and want to keep it framework 4.8 for now.

Should probably try Mono also that may work fine running it as is?

Closing ticket.
